### ISSUE TITLE
[FR] Default to Open Street Map tiles if no mapbox token given

### ISSUE THREAD

**Issue by swheaton** [voxel51/fiftyone#2250]
### Proposal Summary

If no MapBox token is presented, a simple default map layer such as Open Street Map
tile server should be used instead. This lowers the barrier to entry for users experimenting
with the geolocation feature, as well as provides an option to remain completely within
the open source world.

### Motivation

-   _What is the use case for this feature?_
User does not configure a MapBox token but still is able to use the FiftyOne geolocation
feature with a simple default base map.
-   _Why is this use case valuable to support for FiftyOne users in general?_
Requiring a MapBox account and token raises the barrier to entry for a user
trying out the geolocation features of FiftyOne. It also makes this feature
dependent on a closed solution.
-   _Why is this use case valuable to support for your project(s) or
    organization?_
N/A
-   _Why is it currently difficult to achieve this use case? (please be as
    specific as possible about why related FiftyOne features and components are
    insufficient)_
User is required to create a MapBox account and use that token in order to use
the FiftyOne geolocation functionality. If no token is configured, there is no map.

### What areas of FiftyOne does this feature affect?

-   [x] App: FiftyOne application
-   [ ] Core: Core `fiftyone` Python library
-   [ ] Server: FiftyOne server

### Details

This can be achieved by using the Open Street Map tile server in a custom map style,
a la this js fiddle - https://jsfiddle.net/g1rwx8kp/

However a problem is that mapbox-gl >= 2.0 is more closed than it used to be,
and requires a MapBox token even if not using MapBox tile servers.
https://visgl.github.io/react-map-gl/docs/get-started/mapbox-tokens 

The options are these:
1. Reject this FR and require MapBox account and token always
2. Downgrade mapbox-gl to 1.x and accept having no new features
3. Switch to [maplibre-gl](https://maplibre.org/) which is an OSS fork of
mapbox-gl 1.x that attempts feature parity past mapbox-gl 2.0

### Willingness to contribute

The FiftyOne Community encourages new feature contributions. Would you or
another member of your organization be willing to contribute an implementation
of this feature?

-   [x] Yes. I can contribute this feature independently.
-   [ ] Yes. I would be willing to contribute this feature with guidance from
        the FiftyOne community.
-   [ ] No. I cannot contribute this feature at this time.

**Comment by swheaton** [comment]
I have discussed it a bit privately with @benjaminpkane but decided to bring it to public record for the project

**Comment by brimoor** [comment]
Great idea!

**Comment by swheaton** [comment]
@brimoor @benjaminpkane deferring to you on which approach to take
from my list of options below. My preference would be 3 but that's an entirely
new map package.

We also could potentially make this package an optional import so that we
don't require it if map feature is not used?

> 1. Reject this FR and require MapBox account and token always
> 2. Downgrade mapbox-gl to 1.x and accept having no new features
> 3. Switch to [maplibre-gl](https://maplibre.org/) which is an OSS fork of
mapbox-gl 1.x that attempts feature parity past mapbox-gl 2.0

**Comment by brimoor** [comment]
Hmm okay little bit more involved than I thought/hoped. Looks like we should chat offline

**Comment by swheaton** [comment]
Yes me too, I thought I could push a PR in a half hour but it was not so 😞

**Comment by dscer** [comment]
Hi there! Is this issue still active, or have there been any recent developments? I'm interested in using an OSM basemap as well.

**Comment by swheaton** [comment]
Hi @dscer ! Nope we haven't moved forward with this yet. Not enough benefit/demand signal for the risk so far.

Would you be interested in investigating this? I think looking into maplibre-gl is a decent option, we would just want to make sure it's fully compatible with current mapbox use cases, looks good / the same, no degradations in performance or stability, etc.
